### PR TITLE
Improve error descriptions

### PR DESCRIPTION
**Pull request by mhilbrunner**
Small improvements upon the previous PR, generalizes error messages so they work in more contexts. Also adds some comments.

Takes some inspiration from https://github.com/KOBUGE-Games/godot-logger/blob/master/logger.gd

### CODE REVIEW

File: core/error/error_list.cpp [hunk 1]
```diff
@@ -31,55 +31,55 @@
...
+	"Unavailable", // ERR_UNAVAILABLE
+	"Unconfigured", // ERR_UNCONFIGURED
+	"Unauthorized", // ERR_UNAUTHORIZED
+	"Parameter out of range", // ERR_PARAMETER_RANGE_ERROR
+	"Out of memory", // ERR_OUT_OF_MEMORY
+	"Not found", // ERR_FILE_NOT_FOUND
+	"Bad drive", // ERR_FILE_BAD_DRIVE
+	"Bad path", // ERR_FILE_BAD_PATH
+	"Permission denied", // ERR_FILE_NO_PERMISSION
+	"FIle already in use", // ERR_FILE_ALREADY_IN_USE
```
Comment (pycbouh) [inline core/error/error_list.cpp:45]: A typo :)

File: core/error/error_list.cpp [hunk 2]
```diff
@@ -31,55 +31,55 @@
...
-	"Bug",
-	"Printer on fire",
+	"OK", // OK
+	"Failed", // FAILED
+	"Unavailable", // ERR_UNAVAILABLE
+	"Unconfigured", // ERR_UNCONFIGURED
+	"Unauthorized", // ERR_UNAUTHORIZED
+	"Parameter out of range", // ERR_PARAMETER_RANGE_ERROR
+	"Out of memory", // ERR_OUT_OF_MEMORY
+	"Not found", // ERR_FILE_NOT_FOUND
```
Comment (pycbouh) [inline core/error/error_list.cpp:41]: If this is an example of "generalizes error messages so they work in more contexts", then I'm not sure... The constant is still `ERR_FILE_NOT_FOUND`, so we can't really use it anywhere unrelated to files. Same for the permission denied below (which you didn't touch, but it's similar).

Comment (mhilbrunner) [reply core/error/error_list.cpp:41]: TBH, my plan is to move in the opposite direction.

We should have a general ERR_NOT_FOUND, not a special one that only some file methods return (because then this should be moved to a FileError enum).

Same for the permission one, that can be ERR_UNAUTHENTICATED.

I also tried to keep these short and general, because while having whole sentences with explanations can be nice in some cases, it will read real weird or outright wrong in some cases the errors are currently used/returned.

But thats another PR :)

**Review by YuriSizov — APPROVED** [review]
Looks good. Not sure if this touches the core enough to ask for reduz to review. Probably good as is.

File: core/error/error_list.cpp (hunk 2 above)
Comment (pycbouh) [reply core/error/error_list.cpp]: ^ Resolved by adding mentions of "File" back, and generalizing error messages later when we also change the constants.

**Comment by mhilbrunner** [comment]
Nah, this isn't actually used for much yet, and doesn't touch anywhere else. I'll merge :) Thanks for the review!
